File: nysol_python/take/methods/convTex/db/smp/mitemset.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='mitemset'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
tid,item
T1,C
T1,E
T2,D
T2,E
T2,F
T3,A
T3,B
T3,D
T3,F
T4,B
T4,D
T4,F
T5,A
T5,B
T5,D
T5,E
T6,A
T6,B
T6,D
T6,E
T6,F
'''
db['idatas'].append(data)


data={}
data['name']='dat2.csv'
data['text']='''
tid,item,class
T1,C,cls1
T1,E,cls1
T2,D,cls1
T2,E,cls1
T2,F,cls1
T3,A,cls1
T3,B,cls1
T3,D,cls1
T3,F,cls1
T4,B,cls1
T4,D,cls1
T4,F,cls1
T5,A,cls2
T5,B,cls2
T5,D,cls2
T5,E,cls2
T6,A,cls2
T6,B,cls2
T6,D,cls2
T6,E,cls2
T6,F,cls2
'''
db['idatas'].append(data)

data={}
data['name']='taxo.csv'
data['text']='''
item,taxonomy
A,X
B,X
C,Y
D,Z
E,Z
F,Z
'''
db['idatas'].append(data)


############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
3件以上で出現する頻出アイテム集合を列挙する。
'''
script['sh_code']='''
mitemset.rb  S=3 tid=tid item=item i=dat1.csv O=result
'''
script['py_code']='''
import nysol.take as nt
nt.mitemset(S=3,tid="tid",item="item",i="dat1.csv",O="result").run()
'''
script['odatas']='result/patterns.csv,result/tid_pats.csv'
db['scripts'].append(script)

script={}
script['title']='アイテム集合のサイズに制限を加えた例'
script['comment']='''
出現頻度が3以上で、アイテム集合のサイズ3のパターンを列挙する。
'''
script['sh_code']='''
mitemset.rb S=3 l=3 u=3 tid=tid item=item i=dat1.csv O=result
'''
script['py_code']='''
import nysol.take as nt
nt.mitemset(S=3,l=3,u=3,tid="tid",item="item",i="dat1.csv",O="result").run()
'''
script['odatas']='result/patterns.csv'
db['scripts'].append(script)

script={}
script['title']='飽和集合の列挙例'
script['comment']='''
'''
script['sh_code']='''
mitemset.rb S=3 type=C tid=tid item=item i=dat1.csv O=result
'''
script['py_code']='''
import nysol.take as nt
nt.mitemset(S=3,type="C",tid="tid",item="item",i="dat1.csv",O="result").run()
'''
script['odatas']='result/patterns.csv'
db['scripts'].append(script)

script={}
script['title']='極大集合の列挙例'
script['comment']='''
'''
script['sh_code']='''
mitemset.rb S=3 type=M tid=tid item=item i=dat1.csv O=result4
'''
script['py_code']='''
import nysol.take as nt
nt.mitemset(S=3,type="M",tid="tid",item="item",i="dat1.csv",O="result").run()
'''
script['odatas']='result/patterns.csv'
db['scripts'].append(script)

script={}
script['title']='アイテムの階層分類を使った例'
script['comment']='''
'''
script['sh_code']='''
mitemset.rb S=4 tid=tid item=item i=dat1.csv x=taxo.csv taxo=taxonomy O=result
'''
script['py_code']='''
import nysol.take as nt
nt.mitemset(S=4,tid="tid",item="item",i="dat1.csv",x="taxo.csv",taxo="taxonomy",O="result").run()
'''
script['odatas']='result/patterns.csv'
db['scripts'].append(script)

script={}
script['title']='オリジナルアイテムを階層分類で置換する例'
script['comment']='''
'''
script['sh_code']='''
mitemset.rb S=4 tid=tid item=item i=dat1.csv x=taxo.csv taxo=taxonomy -replaceTaxo O=result6
'''
script['py_code']='''
import nysol.take as nt
nt.mitemset(S=4,tid="tid",item="item",i="dat1.csv",x="taxo.csv",taxo="taxonomy",replaceTaxo=True,O="result").run()
'''
script['odatas']='result/patterns.csv'
db['scripts'].append(script)

script={}
script['title']='顕在パターンの列挙例'
script['comment']='''
'''
script['sh_code']='''
mitemset.rb S=2 tid=tid item=item class=class i=dat2.csv p=0.6 O=result7
'''
script['py_code']='''
import nysol.take as nt
nt.mitemset(S=2,tid="tid",item="item",cls="class",i="dat2.csv",p=0.6,O="result").run()
'''
script['odatas']='result/patterns.csv'
db['scripts'].append(script)

File: olddoc/take/jp/examples/mitemset.rb
#!/usr/bin/env ruby
# coding: utf-8
require "../../../scp/mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
tid,item
T1,C
T1,E
T2,D
T2,E
T2,F
T3,A
T3,B
T3,D
T3,F
T4,B
T4,D
T4,F
T5,A
T5,B
T5,D
T5,E
T6,A
T6,B
T6,D
T6,E
T6,F
EOF
)}

File.open("dat2.csv","w"){|fpw| fpw.write(
<<'EOF'
tid,item,class
T1,C,cls1
T1,E,cls1
T2,D,cls1
T2,E,cls1
T2,F,cls1
T3,A,cls1
T3,B,cls1
T3,D,cls1
T3,F,cls1
T4,B,cls1
T4,D,cls1
T4,F,cls1
T5,A,cls2
T5,B,cls2
T5,D,cls2
T5,E,cls2
T6,A,cls2
T6,B,cls2
T6,D,cls2
T6,E,cls2
T6,F,cls2
EOF
)}

File.open("taxo.csv","w"){|fpw| fpw.write(
<<'EOF'
item,taxonomy
A,X
B,X
C,Y
D,Z
E,Z
F,Z
EOF
)}

############## 例1
title="基本例"
comment=<<'EOF'
3件以上で出現する頻出アイテム集合を列挙する。
EOF
scp=<<'EOF'
more dat1.csv
mitemset.rb  S=3 tid=tid item=item i=dat1.csv O=result1
more result1/patterns.csv
more result1/tid_pats.csv
EOF
run(scp,title,comment)

############## 例2
title="アイテム集合のサイズに制限を加えた例"
comment=<<'EOF'
出現頻度が3以上で、アイテム集合のサイズ3のパターンを列挙する。
EOF
scp=<<'EOF'
mitemset.rb S=3 l=3 u=3 tid=tid item=item i=dat1.csv O=result2
more result2/patterns.csv
EOF
run(scp,title,comment)

############## 例3
title="飽和集合の列挙例"
comment=<<'EOF'
EOF
scp=<<'EOF'
mitemset.rb S=3 type=C tid=tid item=item i=dat1.csv O=result3
more result3/patterns.csv
EOF
run(scp,title,comment)

############## 例4
title="極大集合の列挙例"
comment=<<'EOF'
EOF
scp=<<'EOF'
mitemset.rb S=3 type=M tid=tid item=item i=dat1.csv O=result4
more result4/patterns.csv
EOF
run(scp,title,comment)

############## 例5
title="アイテムの階層分類を使った例"
comment=<<'EOF'
EOF
scp=<<'EOF'
more taxo.csv
mitemset.rb S=4 tid=tid item=item i=dat1.csv x=taxo.csv taxo=taxonomy O=result5
more result5/patterns.csv
EOF
run(scp,title,comment)

############## 例6
title="オリジナルアイテムを階層分類で置換する例"
comment=<<'EOF'
EOF
scp=<<'EOF'
more taxo.csv
mitemset.rb S=4 tid=tid item=item i=dat1.csv x=taxo.csv taxo=taxonomy -replaceTaxo O=result6
more result6/patterns.csv
EOF
run(scp,title,comment)

############## 例7
title="顕在パターンの列挙例"
comment=<<'EOF'
EOF
scp=<<'EOF'
more dat2.csv
mitemset.rb S=2 tid=tid item=item class=class i=dat2.csv p=0.6 O=result7
more result7/patterns.csv
EOF
run(scp,title,comment)

